src/platform/packages/private/kbn-scout-reporting/src/cli/upload_events.ts

```diff
@@ -8,60 +8,134 @@
  */
 
 import fs from 'node:fs';
+import path from 'node:path';
+
 import { Command } from '@kbn/dev-cli-runner';
 import { createFlagError } from '@kbn/dev-cli-errors';
+import { ToolingLog } from '@kbn/tooling-log';
+
 import {
   SCOUT_REPORTER_ES_URL,
   SCOUT_REPORTER_ES_API_KEY,
   SCOUT_REPORTER_ES_VERIFY_CERTS,
+  SCOUT_REPORT_OUTPUT_ROOT,
 } from '@kbn/scout-info';
 import { ScoutReportDataStream } from '../reporting/report/events';
 import { getValidatedESClient } from '../helpers/elasticsearch';
 
+const readFilesRecursively = (directory: string, callback: Function) => {
+  const files = fs.readdirSync(directory);
+  files.forEach((file) => {
+    const filePath = path.join(directory, file);
+    const stat = fs.statSync(filePath);
+    if (stat.isDirectory()) {
+      readFilesRecursively(filePath, callback);
+    } else if (stat.isFile()) {
+      callback(filePath);
+    }
+  });
+};
+
+export const uploadAllEventsFromPath = async (
+  eventLogPath: string,
+  options: {
+    esURL: string;
+    esAPIKey: string;
+    verifyTLSCerts: boolean;
+    log: ToolingLog;
+  }
+) => {
+  // Validate CLI options
+  if (!fs.existsSync(eventLogPath)) {
+    throw createFlagError(`The provided event log path '${eventLogPath}' does not exist.`);
+  }
+
+  // If the provided event log path is a file, ensure it ends with .ndjson
+  if (!fs.statSync(eventLogPath).isDirectory()) {
+    if (!eventLogPath.endsWith('.ndjson')) {
+      throw createFlagError(`The provided event log file '${eventLogPath}' must end with .ndjson.`);
+    }
+  } else {
+    // If it's a directory, ensure it contains at least one .ndjson file
+    const files = fs.readdirSync(eventLogPath);
+    const hasNdjsonFile = files.some((file) => file.endsWith('.ndjson'));
+    if (!hasNdjsonFile) {
+      options.log.warning(
+        `The provided event log directory '${eventLogPath}' does not contain any .ndjson files.`
+      );
+      return;
```

Comment:
Not sure this is the best place to put this TBH as this is only scanning the top-level directory, while a `.ndjson` file could exist in a sub-directory (which `readFilesRecursively` would find. And iterating through directories here already and then later a second time for the actual operation seems a bit of an overhead.

What do you think about e.g. collecting stats on the processed report files and at the very end log a warning of no files have been processed?